manifests/package.pp

```diff
@@ -24,154 +24,163 @@
 #
 class elasticsearch::package {
 
-  Exec {
-    path      => [ '/bin', '/usr/bin', '/usr/local/bin' ],
-    cwd       => '/',
-    tries     => 3,
-    try_sleep => 10,
-  }
-
-  #### Package management
-
-
-  # set params: in operation
-  if $elasticsearch::ensure == 'present' {
-
-    # Create directory to place the package file
-    exec { 'create_package_dir_elasticsearch':
-      cwd     => '/',
-      path    => ['/usr/bin', '/bin'],
-      command => "mkdir -p ${elasticsearch::package_dir}",
-      creates => $elasticsearch::package_dir,
-    }
-
-    file { $elasticsearch::package_dir:
-      ensure  => 'directory',
-      purge   => $elasticsearch::purge_package_dir,
-      force   => $elasticsearch::purge_package_dir,
-      backup  => false,
-      require => Exec['create_package_dir_elasticsearch'],
+  case $::osfamily {
```

Comment:
Any specific reason for this? Why wouldn't the current code be able to handle it?